File: unity/Assets/Scripts/Multiplayer/Games/PlayerListResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Games
{
    /// <summary>
    /// Response containing a list of all players in the game in Unity.
    /// </summary>
    [System.Serializable]
    public class PlayerListResponse : ApiResponse<EPlayerListError>
    {
        /// <summary>
        /// The total number of players in the game.
        /// </summary>
        public int count;

        /// <summary>
        /// List of players with their basic information.
        /// </summary>
        public List<PlayerShort> players = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/games/PlayerListResponse.java
package com.michitai.multiplayer.games;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing a list of all players in the game.
 */
public class PlayerListResponse extends ApiResponse {
    @JsonProperty("count")
    private int count;

    @JsonProperty("players")
    private List<PlayerShort> players;

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public List<PlayerShort> getPlayers() {
        return players;
    }

    public void setPlayers(List<PlayerShort> players) {
        this.players = players;
    }
}
